### ISSUE TITLE
Celestial_Navigation_pi  Trusty build

### ISSUE THREAD

**Issue by rgleason** [OpenCPN/plugins#255]
Updated to frontend2, pathnames sorted out, android builds glshim fixed, so all environments build except Ubuntu Trusty,
which has an error: "call of overloaded ‘isinf(double&)’ is ambiguous
     if(isinf(mind))
                  ^

I have tried several things to fix this but have had no success including #include <cmake> 
Trusty uses a different version of widgets I believe and I don't fully understand how isnan changes with compiler.
However this was helpful https://stackoverflow.com/questions/19022561/porting-isnan-to-c11

**Top of  FixDialog.cpp**

```
#include <wx/wx.h>
#include "CelestialNavigationDialog.h"
#include "FixDialog.h"

#include "ocpn_plugin.h"
#include "Sight.h"

#include <vector>
using namespace std;


```

**Subject Code:**
   points.pop_back();
    all_points.push_front(cpoints);

    if(isinf(mind))    <-----LINE 111  ERROR FixDialog.cpp:111:18: error: call of overloaded ‘isinf(double&)’ is ambiguous
        return NAN;
    return mind;
}

**Build Error Report:**  https://app.circleci.com/pipelines/github/rgleason/celestial_navigation_pi/18/workflows/5f61fe9c-1a8f-4646-9914-153deb98dcc3/jobs/174

```
[ 89%] Building CXX object CMakeFiles/celestial_navigation_pi.dir/src/ClockCorrectionDialog.cpp.o
/usr/bin/c++   -DDEBUG_BUILD -DPI_JSON_SCHEMA_VALIDATOR -DTIXML_USE_STL -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -Dcelestial_navigation_pi_EXPORTS -DocpnUSE_GL -I/home/circleci/project/libs/GL -I/home/circleci/project/libs -I/home/circleci/project/api-16 -I/home/circleci/project/src/tinyxml -I/home/circleci/project/src/geomag -I/home/circleci/project/src/astrolabe -I/home/circleci/project/src -I/home/circleci/project/extinclude -I/home/circleci/project/ocpninclude -I/home/circleci/project/extinclude/json-schema-validator -I/home/circleci/project/build/CMakeFiles/include -I/home/circleci/project/include -isystem /usr/lib/x86_64-linux-gnu/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include  -pthread -g -fPIC   -Wall -Wno-unused -fexceptions -rdynamic -fvisibility=hidden  -fno-strict-aliasing  -O0 -g  -DPREFIX=\"/usr/local\" -std=gnu++11 -o CMakeFiles/celestial_navigation_pi.dir/src/ClockCorrectionDialog.cpp.o -c /home/circleci/project/src/ClockCorrectionDialog.cpp
/home/circleci/project/src/FixDialog.cpp: In function ‘double MinCirclePoints(double&, double&, std::list<std::list<wxRealPoint> >&, std::vector<wxRealPoint>&)’:
/home/circleci/project/src/FixDialog.cpp:111:18: error: call of overloaded ‘isinf(double&)’ is ambiguous
     if(isinf(mind))
                  ^
/home/circleci/project/src/FixDialog.cpp:111:18: note: candidates are:
In file included from /usr/include/features.h:374:0,
                 from /usr/include/assert.h:35,
                 from /usr/include/wx-3.0/wx/debug.h:14,
                 from /usr/include/wx-3.0/wx/defs.h:688,
                 from /usr/include/wx-3.0/wx/wx.h:14,
                 from /home/circleci/project/src/FixDialog.cpp:28:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:201:1: note: int isinf(double)
 __MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/wx-3.0/wx/math.h:58:0,
                 from /usr/include/wx-3.0/wx/gdicmn.h:23,
                 from /usr/include/wx-3.0/wx/event.h:20,
                 from /usr/include/wx-3.0/wx/wx.h:24,
                 from /home/circleci/project/src/FixDialog.cpp:28:
/usr/include/c++/4.8/cmath:608:3: note: constexpr bool std::isinf(long double)
   isinf(long double __x)
   ^

```

**Comment by rgleason** [comment]
This trusty build seems to be using wxwidgets 3.0
I wonder if the error -  overloaded ‘isinf(double&)’  would be fixed if trusty was forced to use wxwidgets 3.0.4?

**Comment by dominig** [comment]
I have the same problem with OpenSUSE and the proposed patch works fine.

**Comment by rgleason** [comment]
dominig, where is the proposed patch? I might try it.

**Comment by dominig** [comment]
Here is the patch that use for climatology_pi.

diff --git a/src/ClimatologyOverlayFactory.cpp b/src/ClimatologyOverlayFactory.cpp
index ebbe96e..829e67a 100644
--- a/src/ClimatologyOverlayFactory.cpp
+++ b/src/ClimatologyOverlayFactory.cpp
@@ -42,6 +42,11 @@
 #include "GLES2/gl2.h"
 #endif
 
+/* GTK3 GL include bug */
+#if defined(__WXGTK__)
+#include <GL/glx.h>
+#endif
+
 #include "climatology_pi.h"
 #include "gldefs.h"
 #include "icons.h"

**Comment by rgleason** [comment]
Thank you Dominig!

**Comment by rgleason** [comment]
Dominig,  there appears to be no comparable file called 
celesstial_navigationOverlayFactory.cpp  so I don't quite know where to add these includes for the Trusty build.

**Comment by dominig** [comment]
Oups, Sorry I miss read your issue, appologie my patch is useless.
On my side I have not issue to compile the latest release of that plugin (2.3.2).
I had similar time of issue when I tryed to use the slt version of wx-widgets as you do.
Since I have stcked to the no stl version, those type of issue have gone away.

I use the following dependency (naming respect OpenSUSE but is pretty similar with all distro). I hope that it can help.
BuildRequires: cmake
BuildRequires: gcc-c++
BuildRequires: libstdc++-devel
BuildRequires: gtk3-devel
BuildRequires: wxWidgets-3_2-nostl-devel
BuildRequires: lsb-release
BuildRequires: openSUSE-release
BuildRequires: gettext-runtime
BuildRequires: gettext-tools
BuildRequires: libbz2-devel
BuildRequires: gpsd-devel
BuildRequires: portaudio-devel
BuildRequires: tinyxml-devel
BuildRequires: rpm-build
